### ISSUE TITLE
Authentication issues (starting 03-06-24) preventing Homebridge startup

### ISSUE THREAD

**Issue by jostrasser** [skrollme/homebridge-eveatmo#81]
**Describe the bug**
Netatmo token authentication is not persistent anymore starting 03th of June 2024.

If you restart Homebridge the plugin won´t start anymore.
Recreating the tokens is fixing it temporarily to get Homebridge up one time, after another Homebridge restart you will run into the same issue again and you have to recreate the token again to get Homebridge up and running.

I can see that the tokens are refreshed and cached in the `netatmo-token.json`, but a restart of the HB service invalidates the refresh token.

I requested an update from the Netatmo DEV Support if they made another change on the API (after the well known OAuth2 change on 29th of May 2024). I´ll keep you posted about it.

**Version you use**
- homebridge version 1.8.2
- node version 20.14.0
- plugin version 1.1.1

**To Reproduce**
1. Restart Homebridge service
2. See error (invalid token)

**Comment by jostrasser** [comment]
Hi @skrollme 

got now the feedback, the OAuth2 change was done yesterday, not on the 29th of May (as initially communicated):

> Hello Johannes,
>  
> We just did a modification on the token retrieval process :
>  
> When you refreshed an access_token using the associated endpoint [https://api.netatmo.com/oauth2/token, Netatmo servers](https://api.netatmo.com/oauth2/token,%C2%A0Netatmo%C2%A0servers) responded with a couple of tokens : an access_token and a refresh_token. If the previous access_token was still valid, the refresh_token value was never renewed
>  
> Starting from yesterday, this behavior changed to be compliant with the recommendations of the RFC of the OAuth2 Authorization Framework (section 10.4) and improving the security of the data of our users
> When refreshing tokens, access_token and refresh_token values will be automatically renewed and former tokens invalidated
>  
> So, if you do not update and use the new refresh_token value when refreshing your access_token, your users will be disconnected after 3 hours and you will retrieve an “invalid_grant” error
> To fix it, you need to update the tokens value as soon as you get the newly generated ones
>  
> Please note that we are not linked to third-party apps or to their developments. It’s up to them to follow our token renewal process 
>  
> Have a good day,
> Leslie - Community Manager

There is maybe adoption needed on the plugin side.

Thanks and BR/JO

**Comment by jostrasser** [comment]
Another update, but not relevant here I think:

> Thanks for forwarding the info ! Indeed I asked the teams to add in this deployment also the possibility to have multi-sessions (otherwise, any other connection would have reset the tokens and so making all devices unmanageable via the API). So it took a bit of time for this development 
>  
> Have a good day,
> Leslie - Community Manager

**Comment by mrmaximas** [comment]
> Another update, but not relevant here I think:
> 
> > Thanks for forwarding the info ! Indeed I asked the teams to add in this deployment also the possibility to have multi-sessions (otherwise, any other connection would have reset the tokens and so making all devices unmanageable via the API). So it took a bit of time for this development
> > Have a good day,
> > Leslie - Community Manager

what's stopping you from add a second, third, etc. app to https://dev.netatmo.com/apps?

**Comment by mrmaximas** [comment]
> @mrmaximas I am doing this also. As mentioned, I think this is not relevant here.
> 
> But it is a fact that Homebridge will not survive a restart at the moment.

how you temporary fix this problem after restart?

**Comment by jostrasser** [comment]
> > @mrmaximas I am doing this also. As mentioned, I think this is not relevant here.
> > But it is a fact that Homebridge will not survive a restart at the moment.
> 
> how you temporary fix this problem after restart?

Generating a fresh access and refresh token before you restart, this is the only way right now.

**Comment by mrmaximas** [comment]
> > > @mrmaximas I am doing this also. As mentioned, I think this is not relevant here.
> > > But it is a fact that Homebridge will not survive a restart at the moment.
> > 
> > 
> > how you temporary fix this problem after restart?
> 
> Generating a fresh access and refresh token before you restart, this is the only way right now.

yes, i have issue too after restart
[6/4/2024, 4:00:10 PM] [Netatmo] WARN - Netatmo: Error: getHealthyHomeCoachData error: Invalid access token
[6/4/2024, 4:00:10 PM] [Netatmo] WARN - Netatmo: Error: getStationsDataError error: Invalid access token

**Comment by jostrasser** [comment]
@mrmaximas 

1) generate fresh tokens
2) put them in the json file directly
3) put the refresh key into the plugin configuration
4) restart homebridge

This should fix it until the next restart.

**Comment by mrmaximas** [comment]
after hb restart i just:
1 generate new tokens
2 delete netatmo-token.json file
3 put only one new refresh key into the plugin config
4 restart plugin

**Comment by jostrasser** [comment]
Fine, also a way to go for now :) Thanks

**Comment by mrmaximas** [comment]
> Fine, also a way to go for now :) Thanks

somehow the problem exists 🤗

**Comment by skrollme** [comment]
Hello everyone 
and first of all thanks for your efforts so far, @jostrasser . I just wanted to test by myself, what happens when I restart homebridge, but my setup survived the restart and booted to an up-and-running state again. 
I'm also using the `grant_type=refresh_token` method with a single-purpose app (only for homebridge) at dev.netatmo.com.

**Comment by jostrasser** [comment]
> Hello everyone 
> 
> and first of all thanks for your efforts so far, @jostrasser . I just wanted to test by myself, what happens when I restart homebridge, but my setup survived the restart and booted to an up-and-running state again. 
> 
> I'm also using the `grant_type=refresh_token` method with a single-purpose app (only for homebridge) at dev.netatmo.com.

Hi @skrollme 
U're welcome, as always ;)

This is strange because I got the same errors yesterday evening like @mrmaximas : Access and Refresh token invalid...

I am also using a separate "App" at dev.netatmo.com for the plugin, the same grant_type... so there is no difference I think.

@mrmaximas hot is it at the moment on your side? Is a restart working for you now?

**Comment by palasinio** [comment]
Have you tried to restart Homebridge twice?Am 04.06.2024 um 21:19 schrieb Johannes Strasser ***@***.***>:﻿

Hello everyone
and first of all thanks for your efforts so far, @jostrasser . I just wanted to test by myself, what happens when I restart homebridge, but my setup survived the restart and booted to an up-and-running state again.
I'm also using the grant_type=refresh_token method with a single-purpose app (only for homebridge) at dev.netatmo.com.

Hi @skrollme
U're welcome, as always ;)
This is strange because I got the same errors yesterday evening like @mrmaximas : Access and Refresh token invalid...
I am also using a separate "App" at dev.netatmo.com for the plugin, the same grant_type... so there is no difference I think.
@mrmaximas hot is it at the moment on your side? Is a restart working for you now?

**Comment by PvdGulik** [comment]
Just a restart of Homebridge now stopped Eveatmo working, due to a  Invalid access token

`
[05/06/2024, 20:58:42] [eveatmo platform] Loading new data from API for: weatherstation
[05/06/2024, 21:01:43] [Homebridge UI] Running Command: /opt/homebridge/lib/node_modules/homebridge-config-ui-x/plugin-upgrade-install.sh homebridge-lg-thinq 1.8.5 /var/lib/homebridge/node_modules
[05/06/2024, 21:01:55] [Homebridge UI] Homebridge restart request received
[05/06/2024, 21:01:55] [Homebridge UI] UI / Bridge settings have not changed; only restarting Homebridge process
[05/06/2024, 21:01:55] [Homebridge UI] Sending SIGTERM to Homebridge
[05/06/2024, 21:01:55] Got SIGTERM, shutting down Homebridge...
[05/06/2024, 21:02:00] [HB Supervisor] Homebridge Process Ended. Code: 143, Signal: null
[05/06/2024, 21:02:05] [HB Supervisor] Restarting Homebridge...
[05/06/2024, 21:02:05] [HB Supervisor] Starting Homebridge with extra flags: -I -P /var/lib/homebridge/node_modules --strict-plugin-resolution
[05/06/2024, 21:02:05] [HB Supervisor] Started Homebridge v1.8.2 with PID: 6259
[05/06/2024, 21:02:08] Loaded config.json with 3 accessories and 12 platforms.
[05/06/2024, 21:02:08] Loaded 10 cached accessories from cachedAccessories.
[05/06/2024, 21:02:09] ---
[05/06/2024, 21:02:10] ---
[05/06/2024, 21:02:10] ---
[05/06/2024, 21:02:11] ---
[05/06/2024, 21:02:12] Registering platform 'homebridge-eveatmo.eveatmo'
[05/06/2024, 21:02:12] ---
[05/06/2024, 21:02:15] ---
[05/06/2024, 21:02:16] ---
[05/06/2024, 21:02:16] ---
[05/06/2024, 21:02:18] ---
[05/06/2024, 21:02:18] ---
[05/06/2024, 21:02:24] ---
[05/06/2024, 21:02:24] ---
[05/06/2024, 21:02:25] ---
[05/06/2024, 21:02:26] ---
[05/06/2024, 21:02:27] ---
[05/06/2024, 21:02:36] ---
[05/06/2024, 21:02:36] Loading 12 platforms...
[05/06/2024, 21:02:36] [eveatmo platform] Initializing eveatmo platform...
[05/06/2024, 21:02:36] [eveatmo platform] Authenticating using 'refresh_token' grant
[05/06/2024, 21:02:44] [eveatmo platform] Loading new data from API for: weatherstation
[05/06/2024, 21:02:46] [eveatmo platform] WARN - Netatmo: Error: getStationsDataError error: Invalid access token
`

**Comment by skrollme** [comment]
> Have you tried to restart Homebridge twice
Just did another two restarts in a row, @palasinio . Still boots up and works like expected. But I will take a look at this soon.

But just to understand what is happening: Restarting twice (in a row) leads to error messages for you. But when you delete the `netatmo-token.json` and restart it starts working again. Ist this correct?

**Comment by mrmaximas** [comment]
> Have you tried to restart Homebridge twice?Am 04.06.2024 um 21:19 schrieb Johannes Strasser ***@***.***>:﻿ Hello everyone and first of all thanks for your efforts so far, @jostrasser . I just wanted to test by myself, what happens when I restart homebridge, but my setup survived the restart and booted to an up-and-running state again. I'm also using the grant_type=refresh_token method with a single-purpose app (only for homebridge) at dev.netatmo.com. Hi @skrollme U're welcome, as always ;) This is strange because I got the same errors yesterday evening like @mrmaximas : Access and Refresh token invalid... I am also using a separate "App" at dev.netatmo.com for the plugin, the same grant_type... so there is no difference I think. @mrmaximas hot is it at the moment on your side? Is a restart working for you now? —Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

Sorry for this delay, but it's magic. i just restarted the plugin and everything works without errors.

[6/6/2024, 10:03:46 AM] [Netatmo] Restarting child bridge...
[6/6/2024, 10:03:46 AM] Got SIGTERM, shutting down child bridge process...
[6/6/2024, 10:03:51 AM] [Netatmo] Child bridge process ended
[6/6/2024, 10:03:51 AM] [Netatmo] Process Ended. Code: 143, Signal: null
[6/6/2024, 10:03:58 AM] [Netatmo] Restarting Process...
[6/6/2024, 10:03:58 AM] [Netatmo] Launched child bridge with PID 13476
[6/6/2024, 10:03:59 AM] Registering platform 'homebridge-eveatmo.eveatmo'
[6/6/2024, 10:03:59 AM] [Netatmo] Loaded homebridge-eveatmo v1.1.1 child bridge successfully
[6/6/2024, 10:03:59 AM] Loaded 0 cached accessories from cachedAccessories.0E34C832B317.
[6/6/2024, 10:03:59 AM] [Netatmo] Authenticating using 'refresh_token' grant
[6/6/2024, 10:03:59 AM] [Netatmo] Loading new data from API for: weatherstation
[6/6/2024, 10:03:59 AM] [Netatmo] Loading new data from API for: airquality
[6/6/2024, 10:03:59 AM] Initializing platform accessory 'LIVING ROOM -'...
[6/6/2024, 10:03:59 AM] Initializing platform accessory 'BEDROOM -'...
[6/6/2024, 10:03:59 AM] Initializing platform accessory 'GUEST ROOM -'...
[6/6/2024, 10:03:59 AM] Initializing platform accessory 'OUTDOOR -'...
[6/6/2024, 10:03:59 AM] Initializing platform accessory 'KITCHEN -'...
[6/6/2024, 10:04:00 AM] Homebridge v1.8.2 (HAP v0.12.1) (Netatmo) is running on port 44273.

**Comment by jostrasser** [comment]
@skrollme 
This is strange... I restarted now and I am running into the same behavior:

```
[06/06/2024, 10:58:20] [eveatmo-platform] WARN - Netatmo: Error: getHealthyHomeCoachData error: Invalid access token
[06/06/2024, 10:58:20] [eveatmo-platform] WARN - Netatmo: Error: getStationsDataError error: Invalid access token
[06/06/2024, 10:58:37] [homebridge-eveatmo] This plugin is taking long time to load and preventing Homebridge from starting. See https://homebridge.io/w/JtMGR for more info.
[06/06/2024, 10:58:57] [homebridge-eveatmo] This plugin is taking long time to load and preventing Homebridge from starting. See https://homebridge.io/w/JtMGR for more info.
[06/06/2024, 10:59:17] [homebridge-eveatmo] This plugin is taking long time to load and preventing Homebridge from starting. See https://homebridge.io/w/JtMGR for more info.
[06/06/2024, 10:59:37] [homebridge-eveatmo] This plugin is taking long time to load and preventing Homebridge from starting. See https://homebridge.io/w/JtMGR for more info.
[06/06/2024, 10:59:57] [homebridge-eveatmo] This plugin is taking long time to load and preventing Homebridge from starting. See https://homebridge.io/w/JtMGR for more info.
[06/06/2024, 11:00:17] [homebridge-eveatmo] This plugin is taking long time to load and preventing Homebridge from starting. See https://homebridge.io/w/JtMGR for more info.
```

I restarted twice, same behavior.

Now I tried:
1) removed the netatmo-token.json
2) created and changed the refresh token to a new one in the plugin config
3) Homebridge is starting
4) restarted the Homebridge again
5) Homebridge stucking at the same point: Invalid access token

So rebooting twice is not working after a manual token refresh.

If I only remove the json and restart:
[06/06/2024, 11:12:10] [eveatmo-platform] WARN - Netatmo: Error: Authenticate refresh error: invalid_grant

Config:

```
{
            "name": "eveatmo-platform",
            "ttl": 300,
            "extra_aq_sensor": true,
            "extra_co2_sensor": true,
            "weatherstation": true,
            "airquality": true,
            "co2_alert_threshold": 4990,
            "log_info_msg": true,
            "auth": {
                "client_id": "62f95941a4f94xxxxxxxxxxxxxx",
                "client_secret": "Oc3OvrHHwshnSsq1WEBylwpRxxxxxxxxxxxxxxx",
                "grant_type": "refresh_token",
                "refresh_token": "56487281c6aa42d641xxxxxx|0ddec8c6ac2f1f561eca56265xxxxxxxx"
            },
            "platform": "eveatmo"
        },
```

What can be my problem here?

**Comment by jostrasser** [comment]
@mrmaximas thanks for the update, I think this information is essential.

I always restarted the whole Homebridge service or system, never the plugin only.
I don´t use child bridges, all plugins are running on the main bridge.

**Comment by skrollme** [comment]
Sounds like this will become a fun debugging session *yikes*

**Comment by Kositch** [comment]
Same issue by me - after complete HB restart I need to regenerate new tokens, until the bridge states Invalid Access Token.

**Comment by ro10dev** [comment]
Having the same issue. Generating new tokens only temporarily fixes the problem until the next restart.

**Comment by PvdGulik** [comment]
> Just a restart of Homebridge now stopped Eveatmo working, due to a Invalid access token.
> 
> [05/06/2024, 20:58:42] [eveatmo platform] Loading new data from API for: weatherstation [05/06/2024, 21:01:43] [Homebridge UI] Running Command: /opt/homebridge/lib/node_modules/homebridge-config-ui-x/plugin-upgrade-install.sh homebridge-lg-thinq 1.8.5 /var/lib/homebridge/node_modules [05/06/2024, 21:01:55] [Homebridge UI] Homebridge restart request received [05/06/2024, 21:01:55] [Homebridge UI] UI / Bridge settings have not changed; only restarting Homebridge process [05/06/2024, 21:01:55] [Homebridge UI] Sending SIGTERM to Homebridge [05/06/2024, 21:01:55] Got SIGTERM, shutting down Homebridge... [05/06/2024, 21:02:00] [HB Supervisor] Homebridge Process Ended. Code: 143, Signal: null [05/06/2024, 21:02:05] [HB Supervisor] Restarting Homebridge... [05/06/2024, 21:02:05] [HB Supervisor] Starting Homebridge with extra flags: -I -P /var/lib/homebridge/node_modules --strict-plugin-resolution [05/06/2024, 21:02:05] [HB Supervisor] Started Homebridge v1.8.2 with PID: 6259 [05/06/2024, 21:02:08] Loaded config.json with 3 accessories and 12 platforms. [05/06/2024, 21:02:08] Loaded 10 cached accessories from cachedAccessories. [05/06/2024, 21:02:09] --- [05/06/2024, 21:02:10] --- [05/06/2024, 21:02:10] --- [05/06/2024, 21:02:11] --- [05/06/2024, 21:02:12] Registering platform 'homebridge-eveatmo.eveatmo' [05/06/2024, 21:02:12] --- [05/06/2024, 21:02:15] --- [05/06/2024, 21:02:16] --- [05/06/2024, 21:02:16] --- [05/06/2024, 21:02:18] --- [05/06/2024, 21:02:18] --- [05/06/2024, 21:02:24] --- [05/06/2024, 21:02:24] --- [05/06/2024, 21:02:25] --- [05/06/2024, 21:02:26] --- [05/06/2024, 21:02:27] --- [05/06/2024, 21:02:36] --- [05/06/2024, 21:02:36] Loading 12 platforms... [05/06/2024, 21:02:36] [eveatmo platform] Initializing eveatmo platform... [05/06/2024, 21:02:36] [eveatmo platform] Authenticating using 'refresh_token' grant [05/06/2024, 21:02:44] [eveatmo platform] Loading new data from API for: weatherstation [05/06/2024, 21:02:46] [eveatmo platform] WARN - Netatmo: Error: getStationsDataError error: Invalid access token
> 
> A seccond restart did not solve the problem...
> 
> Update: removing netatmo-token.json did, creating new refresh-token, add in config, and restart Hombridge did the trick

After a 4 days with 0 problems, I restarted Hombridge, and the problem is back.

Just removing netatmo-token.json again, creating new refresh-token, add in config, and restart Hombridge makes it working again.

Would be nice that is would work after a restart without the need to remove the netamo-token.json and recreate a refresh-token.

Thanks!

**Comment by t0mmey** [comment]
I have the same issue. So happy I found this thread. After restarting Homebridge (Hoobs 4) today, the bridge didn't come back up.

I am using Hoobs and I can't even find the netamo-token.json in /home/hoobs/.hoobs/etc/. At least this is where the config.json is stored. Any idea where I can find the netamo-token.json file? Or can I just create the file? And if so, what exactly need to go into the file?

**Comment by jostrasser** [comment]
> I am using Hoobs and I can't even find the netamo-token.json in /home/hoobs/.hoobs/etc/. At least this is where the config.json is stored. Any idea where I can find the netamo-token.json file? Or can I just create the file? And if so, what exactly need to go into the file?

A native Homebridge installation saves it under `/var/lib/homebridge`. I think you can find it maybe under `/var/lib/hoobs`.

**Comment by t0mmey** [comment]
> > I am using Hoobs and I can't even find the netamo-token.json in /home/hoobs/.hoobs/etc/. At least this is where the config.json is stored. Any idea where I can find the netamo-token.json file? Or can I just create the file? And if so, what exactly need to go into the file?
> 
> A native Homebridge installation saves it under `/var/lib/homebridge`. I think you can find it maybe under `/var/lib/hoobs`.

Thanks a lot mate. Found it under /var/lib/hoobs/hoobs. Deleted it, generated a new token and it works again (until the next reboot, just tested it).

**Comment by Kositch** [comment]
I don't need to delete anything, generating new token and putting it in the plugin config is enough to start it working.

**Comment by t0mmey** [comment]
For me it's different - if I generate a new token only I still get the Invalid access token error message during start of homebridge. Only if I delete the netamo-token.json file in addition it starts working again

**Comment by mrmaximas** [comment]
> For me it's different - if I generate a new token only I still get the Invalid access token error message during start of homebridge. Only if I delete the netamo-token.json file in addition it starts working again

i confirm the same behaviour - only after delete the netamo-token.json it starts working again

**Comment by jostrasser** [comment]
Same behavior like @mrmaximas and @t0mmey 
Without the deletion of the netatmo-token.json Homebridge will not start.

**Comment by Kositch** [comment]
> Same behavior like @mrmaximas and @t0mmey Without the deletion of the netatmo-token.json Homebridge will not start (and a new key for sure).
> 
> Strange that there are users with a different behavior. Not really consistent 🤔

That is interesting, I have this plugin in child bridge, mayby the reason?

**Comment by mrmaximas** [comment]
> That is interesting, I have this plugin in child bridge, mayby the reason?

FYI - i use all of my HB plugins via bridges too.

**Comment by PvdGulik** [comment]
> > For me it's different - if I generate a new token only I still get the Invalid access token error message during start of homebridge. Only if I delete the netamo-token.json file in addition it starts working again
> 
> i confirm the same behaviour - only after delete the netamo-token.json + put new update key it starts working again

Same as I decribed.. https://github.com/skrollme/homebridge-eveatmo/issues/81#issuecomment-2156750851

**Comment by PvdGulik** [comment]
> Same as I decribed.. [#81 (comment)](https://github.com/skrollme/homebridge-eveatmo/issues/81#issuecomment-2156750851)

Seems like same as I decribed.. https://github.com/skrollme/homebridge-eveatmo/issues/81#issuecomment-2156750851

**Comment by skrollme** [comment]
I just released a new version [1.1.2](https://github.com/skrollme/homebridge-eveatmo/releases/tag/v1.1.2) which hopefully fixes the startup problems. 🤞 

It looks like it was some kind of race-condition where an already existing _access_token_ (loaded from _netamo-token.json_) got invalidated (by a background refresh) just before it should be used. Solution: Do not store the _access_token_ in the _netamo-token.json_ at all 🤷 

If you still have a working _refresh_token_ in your _netamo-token.json_, just just have to update the plugin and everything should be fine.

**Comment by mrmaximas** [comment]
> I just released a new version [1.1.2](https://github.com/skrollme/homebridge-eveatmo/releases/tag/v1.1.2) which hopefully fixes the startup problems. 🤞
> 
> It looks like it was some kind of race-condition where an already existing _access_token_ (loaded from _netamo-token.json_) got invalidated (by a background refresh) just before it should be used. Solution: Do not store the _access_token_ in the _netamo-token.json_ at all 🤷
> 
> If you still have a working _refresh_token_ in your _netamo-token.json_, just just have to update the plugin and everything should be fine.

It's works after full HB restart! Thanks for you kind support!

**Comment by jostrasser** [comment]
Thanks @skrollme for the quick fix! 🥳

**Comment by PvdGulik** [comment]
Problem solved! Thanks!

> [11/06/2024, 22:30:18] [eveatmo platform] Loading new data from API for: weatherstation
> [11/06/2024, 22:39:18] [eveatmo platform] Loading new data from API for: weatherstation
> [11/06/2024, 22:39:59] [Homebridge UI] Running Command: npm install --save homebridge-eveatmo@1.1.2
> [11/06/2024, 22:40:25] [Homebridge UI] Homebridge restart request received
> [11/06/2024, 22:40:25] [Homebridge UI] UI / Bridge settings have not changed; only restarting Homebridge process
> [11/06/2024, 22:40:25] [Homebridge UI] Sending SIGTERM to Homebridge
> [11/06/2024, 22:40:25] Got SIGTERM, shutting down Homebridge...
> [11/06/2024, 22:40:30] [HB Supervisor] Homebridge Process Ended. Code: 143, Signal: null
> [11/06/2024, 22:40:35] [HB Supervisor] Restarting Homebridge...
> [11/06/2024, 22:40:35] [HB Supervisor] Starting Homebridge with extra flags: -I -P /var/lib/homebridge/node_modules --strict-plugin-resolution
> [11/06/2024, 22:40:35] [HB Supervisor] Started Homebridge v1.8.2 with PID: 24205
> [11/06/2024, 22:40:38] Loaded config.json with 3 accessories and 12 platforms.
> [11/06/2024, 22:40:38] Loaded 10 cached accessories from cachedAccessories.
> 
> [11/06/2024, 22:40:40] ---
> [11/06/2024, 22:40:41] Loaded plugin: homebridge-eveatmo@1.1.2
> [11/06/2024, 22:40:41] Registering platform 'homebridge-eveatmo.eveatmo'
> [11/06/2024, 22:40:41] ---
> 
> [11/06/2024, 22:41:00] [eveatmo platform] Initializing eveatmo platform...
> [11/06/2024, 22:41:00] [eveatmo platform] Authenticating using 'refresh_token' grant
> [11/06/2024, 22:41:06] [eveatmo platform] Loading new data from API for: weatherstation
> 
> [11/06/2024, 22:41:08] Initializing platform accessory 'Woning (Binnenmodule) Binnenmodule'...
> [11/06/2024, 22:41:08] Initializing platform accessory 'Woning (Binnenmodule) Buitenmodule'...
> 
> [11/06/2024, 22:50:06] [eveatmo platform] Loading new data from API for: weatherstation
> 
> [11/06/2024, 22:59:06] [eveatmo platform] Loading new data from API for: weatherstation

Thanks for the quick fix!

**Comment by glentregoning** [comment]
For others that suffered this issue - I still needed to manually jump into terminal and delete netatmo-token.json in order to fix this issue after upgrading to 1.1.2.

**Comment by skrollme** [comment]
> For others that suffered this issue - I still needed to manually jump into terminal and delete netatmo-token.json in order to fix this issue after upgrading to 1.1.2.

Thanks for the additional information, @glentregoning . But that should only be the case, if your _refresh_token_ has also expired.
